### PR TITLE
feat: Cache remote config

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

We use a CDN for the public endpoints but there are multiple reasons to cache the remote configs oursevles:
1. We want to backport /decide to use this eventually instead of the TeamCache
2. Self hosted still wants fast caching
3. When hitting source we don't want to have issues due to postgres unavailability for some reason

## Changes

* Adds new methods that cache the result when accessed, including setting these values when syncs happen

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

File: posthog/api/remote_config.py [hunk 1]
```diff
@@ -11,28 +13,38 @@ class BaseRemoteConfigAPIView(APIView):
...
     permission_classes = []
 
-    def get_object(self, token: str) -> RemoteConfig:
-        try:
-            return RemoteConfig.objects.get(team__api_token=token)
-        except RemoteConfig.DoesNotExist:
-            raise Http404()
+    def check_token(self, token: str):
+        # Check token is a valid token - not longer than 255 characters and only allowing alphanumeric characters and _
+        if len(token) > 200 or not re.match(r"^[a-zA-Z0-9_]+$", token):
```
Comment (mariusandra) [inline posthog/api/remote_config.py:18]: `255` != `200`

**Review by mariusandra — APPROVED** [review]
Looks good. Question though: will this suddenly make Redis's memory usage explode? Roughly 100kb per array, over 100k users = 9GB. I think that's fine, but we should monitor this, or possibly gzip the data before storing it.

File: posthog/api/remote_config.py (hunk 1 above)
Comment (benjackwhite) [reply posthog/api/remote_config.py:18]: Ooh yeah comment copy paste error.

For the configs size, we dont store arrayjs - only the config js side of things. So average team config is only a couple of kb max. Could for sure compress it if it seems to be an issue but we already store the team configs in redis via the TeamCache serializer so I don't expect it to be much different.

Comment (benjackwhite) [reply posthog/api/remote_config.py:18]: I'll add some metric counters to track this though

**Pull request merged**